c10/xpu/XPUCachingAllocator.h

```diff
@@ -5,6 +5,13 @@
 #include <c10/xpu/XPUStream.h>
 
 namespace c10::xpu::XPUCachingAllocator {
+class XPUAllocator : public DeviceAllocator {
+ public:
+  virtual void init(DeviceIndex device_count) = 0;
```

Comment:
```suggestion
```
Maybe we can keep things simple for now. Since we don’t currently see a real use case for the init function, we can leave it out and add it back once we actually need it.

Reply:
```suggestion
```
ditto